pkg/metrics/metrics.go

```diff
@@ -182,6 +184,14 @@ var (
 		[]string{Outcome},
 	)
 
+	installPlanWarningCount = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "install_plan_warnings_count",
+			Help: "monotonic count of resources created with warnings (for example, deprecation warnings)",
+		},
+		[]string{WARNING_LABEL, NAME_LABEL, NAMESPACE_LABEL, GVK_LABEL},
```

Comment:
It's certainly a concern, but since we are just alerting here on deprecated objects on creation I'm wondering where the cardinality would truly explode? Is it cases where the installplan includes additional subscriptions from dependency resolution and you basically have an N dimensional install 

I was looking to embed things like the installplan name or even better the operator package in the WarningRoundTripper struct, instantiate it in the `ExcutePlan()` step with all the info, and then pass it down when creating the clients